### PR TITLE
feat(kspm-collector): Deployment annotations and fix label render issue

### PR DESCRIPTION
**Pull request by narendramannam**
## Checklist

- [x] Title of the PR starts with type and scope, (e.g. `feat(agent,node-analyzer,sysdig-deploy):`)
- [x] Chart Version bumped for the respective charts
- [x] Variables are documented in the README.md (or README.tpl in some charts)
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [x] All test files are added in the tests folder of their respective chart and have a "_test" suffix

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

File: charts/kspm-collector/templates/deployment.yaml [hunk 1]
```diff
@@ -16,11 +20,11 @@ spec:
     metadata:
       labels:
         app.kubernetes.io/name: {{ include "kspmCollector.name" . }}
+{{ include "kspmCollector.labels" . | indent 8 }}
```
Comment (mavimo) [inline charts/kspm-collector/templates/deployment.yaml]: Can I suggest to use
```suggestion
        {{ include "kspmCollector.labels" . | nindent 8 }}
```
to make the document format more easy to understand? WDYT?

**Comment by narendramannam** [comment]
@mavimo  Thank you for the feedback. I have implemented the changes you suggested in the code

**Review by chen-shmilovich-sysdig — APPROVED** [review]
LGTM!

**Comment by narendramannam** [comment]
Thanks @chen-shmilovich-sysdig  Can you please approve again, I had to resolve chart version conflicts. Thanks

**Comment by mavimo** [comment]
@narendramannam can you rebase this PR on top of main? Once done I'll take care of merging it. 

PS: there is also a wrong indentation in the `README.md` file in the `annotations` section.
image

**Comment by narendramannam** [comment]
Thanks @mavimo , updated the PR

**Review by mavimo — APPROVED** [review]

**Comment by narendramannam** [comment]
Hi @mavimo  I see the tests fail for one of the chart, but I don't think the changes are related. are you or someone taking a loot at the tests?

**Comment by mavimo** [comment]
@narendramannam if we bump a child chart in `charts/sysdig-deploy/Chart.yaml` we should bump also the chart version for the `sysdig-deploy` chart.

**Comment by narendramannam** [comment]
@mavimo I did bump the chart version but I think the test value render has some issue? https://github.com/sysdiglabs/charts/pull/1626/files#diff-2c335de459a836b7fae09ab7f921d1564833cde0d28d16c1bdc90774e314b305R51

**Comment by mavimo** [comment]
@narendramannam thx, I'm looking into it, I'll keep you posted

BTW: I'm referring to https://github.com/sysdiglabs/charts/pull/1626/files#diff-2c335de459a836b7fae09ab7f921d1564833cde0d28d16c1bdc90774e314b305R5 where we need to upgrade from 1.40.0 to 1.41.0

**Comment by narendramannam** [comment]
@mavimo Thanks for pointing it out, I've updated the chart version now

**Comment by mavimo** [comment]
@narendramannam sorry to bother you again... but it seems that we still have some linting issues as you can see [here](https://github.com/sysdiglabs/charts/actions/runs/8185310626?pr=1626):

image

**Comment by narendramannam** [comment]
Hi @mavimo I couldn't re-produce this locally, belive it's due to the test values substituted has some issue with trailing whitespaces?

**Comment by narendramannam** [comment]
Hi @mavimo  I was able to replicate the `ct` locally and see that when `SECURE_AGENT_TOKEN` properly, the lint succeeds,

```
root@389abf57cefb:/sysdig-charts# yamllint charts/sysdig-deploy/ci/test-default-saas-values.yaml charts/kspm-collector/ci/test-values.yaml
charts/sysdig-deploy/ci/test-default-saas-values.yaml
  1:1       warning  missing document start "---"  (document-start)

charts/kspm-collector/ci/test-values.yaml
  1:1       warning  missing document start "---"  (document-start)

root@389abf57cefb:/sysdig-charts#
```
if `SECURE_AGENT_TOKEN` set to empty,

```
root@389abf57cefb:/sysdig-charts# yamllint charts/kspm-collector/ci/test-values.yaml
charts/kspm-collector/ci/test-values.yaml
  1:1       warning  missing document start "---"  (document-start)
  2:13      error    trailing spaces  (trailing-spaces)
```

Can you please cross-check and help me expedite the PR? We're in middle of testing the cluster upgrades, it would greatly help us meet the timelines as it also contains a fix for `selectors` miss-match issue we're facing

**Comment by mavimo** [comment]
@narendramannam 👍

**Comment by mavimo** [comment]
@narendramannam as I can't push changes in your branch I made a copy with the chart version fix and the rest are running in https://github.com/sysdiglabs/charts/pull/1639

**Comment by mavimo** [comment]
@narendramannam merged in https://github.com/sysdiglabs/charts/pull/1639 I'm going to close this PR

Thanks for your contribution 💪

**Comment by narendramannam** [comment]
Thanks a ton @mavimo , May I ask what had caused the issue so I'll pay attention to these for future PRs

**Comment by mavimo** [comment]
@narendramannam there are two "issues", one is that we bump chart for each PR merged and PR tend to become stale (need rebase, new chart version, ..), this is not always is an issue but sometime when the PR queue pile up it become hard to manage.
The second issue is related to the lack of the token available in CI, we never experienced this in the past and is something that we are still investigating.

You already made an amazing work so is not on you at all! 🙇

**Pull request closed without merging**